lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,486 @@
+# Copyright: (c) 2018, Deric Crago <deric.crago@gmail.com>
+# Copyright: (c) 2018, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import re
+from os.path import dirname, exists, getsize
+from socket import gaierror
+from ssl import SSLEOFError, SSLError
+from time import sleep
+import urllib3
+import traceback
+
+REQUESTS_IMP_ERR = None
+try:
+    import requests
+    HAS_REQUESTS = True
+except ImportError:
+    REQUESTS_IMP_ERR = traceback.format_exc()
+    HAS_REQUESTS = False
```

Comment:
i would avoid using requests, introduced dep that is not needed, as functions in urls.py  should cover what this plugin uses.

Reply:
I agree in principle that the built-in methods should be sufficient, as it's doing one `requests.get` and one `requests.put` ...but I don't know how to translate between the two...

On further inspection, it looks like the built-in urls.py doesn't support chunk_size and piecemeal fetching of remote files, so I'm inclined to just leave the requests dep as-is.

This is only a dep for the controller, not the managed nodes, and all other vmware components in ansible also require python requests

Reply:
Can you please raise an issue for this so we can get the built in updated

Reply:
if its already a pvimomi dependency its ok to use, but then you really dont need to manage it this way, just import it in same block you do pyvimomi and only deal with missing pyvimomi

Reply:
I'll raise an issue.  On my RHEL7 system, requests doesn't seem to be a direct dep of pyvmomi, but on my Fedora 29 system, python3-requests is a direct dep of python3-pyvmomi.

Reply:
Issue opened: https://github.com/ansible/ansible/issues/55162